swap/swap_test.go

```diff
@@ -1818,6 +1821,83 @@ func TestPeerGetLastSentCumulativePayout(t *testing.T) {
 	}
 }
 
+func TestAvailableBalance(t *testing.T) {
+	testBackend := newTestBackend()
+	defer testBackend.Close()
+	swap, clean := newTestSwap(t, ownerKey, testBackend)
+	defer clean()
+	cleanup := setupContractTest()
+	defer cleanup()
+
+	// deploy a chequebook
+	err := testDeploy(context.TODO(), swap, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// create a peer
+	peer, err := swap.addPeer(newDummyPeerWithSpec(Spec).Peer, swap.owner.address, swap.GetParams().ContractAddress)
+
+	// deposit 100
+	depositAmount := big.NewInt(100)
+	opts := bind.NewKeyedTransactor(swap.owner.privateKey)
+	opts.Value = depositAmount
+	opts.Context = context.TODO()
+	rec, err := swap.contract.Deposit(opts, swap.backend)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if rec.Status != types.ReceiptStatusSuccessful {
+		t.Fatal("Transaction reverted")
+	}
+
+	// verify that available balance equals depositAmount
+	availableBalance, err := swap.computeAvailableBalance()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if availableBalance != depositAmount.Uint64() {
+		t.Fatalf("availableBalance not equal to deposited amount. availableBalance: %d, depositAmount: %d", availableBalance, depositAmount.Uint64())
+	}
+	// withdraw 50
+	withdrawAmount := big.NewInt(50)
+	opts = bind.NewKeyedTransactor(swap.owner.privateKey)
+	opts.Context = context.TODO()
+	rec, err = swap.contract.Withdraw(opts, swap.backend, withdrawAmount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if rec.Status != types.ReceiptStatusSuccessful {
+		t.Fatal("Transaction reverted")
+	}
+
+	// verify available balance
+	availableBalance, err = swap.computeAvailableBalance()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if availableBalance != (depositAmount.Uint64() - withdrawAmount.Uint64()) {
```

Comment:
extract this subtraction into a variable

Reply:
Done